### PR TITLE
helm: add value for setting pod annotations

### LINKED ISSUE #435: Helm value for setting pod annotations
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Kubernetes version: (use `kubectl version`):

### PR DESCRIPTION
**Pull request by ZhijunZhao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #435 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ZhijunZhao** [comment]
I've signed the CNCF agreement as IC.

File: manifest_staging/charts/secrets-store-csi-driver/README.md [hunk 1]
```diff
@@ -43,7 +43,8 @@ The following table lists the configurable parameters of the csi-secrets-store-p
 | `linux.livenessProbe.resources`         | The resource request/limits for the linux liveness-probe container image                                                          | `limits: 100m CPU, 100Mi; requests: 10m CPU, 20Mi`  |
 | `linux.env`                             | Environment variables to be passed for the daemonset on linux nodes                                                               | `[]`                                                |
 | `linux.priorityClassName`               | Indicates the importance of a Pod relative to other Pods.                                                                         | `""`                                                |
-| `linux.annotations`                     | Linux *Daemonset* annotations                                                                                                     | `{}`                                                |
+| `linux.daemonSetAnnotations`            | Linux *DaemonSet* annotations                                                                                                     | `{}`                                                |
```
Comment (aramase) [inline manifest_staging/charts/secrets-store-csi-driver/README.md:46]: Let's use `daemonsetAnnotations` instead of `daemonSetAnnotations`

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**